File: Assignment1/fourargs.cpp
//copyright 2018 Zhuyu Li zhuyuli@bu.edu

/*
A C++ program that outputs the first four arguments to
stdout stream and any additional arguments to stderr stream
*/

#include <iostream>

int main(int argc, char* argv[])
{
	if (argc <= 5) {
		for (int i = 1; i <= argc; i++) {
			std::cout << argv[i] << std::endl;
		}
	}
	else {
		for (int i = 1; i < 5; i++) {
			std::cout << argv[i] << std::endl;
		}
		for (int i = 5; i < argc; i++) {
			std::cerr << argv[i] << std::endl;	
		}

	}

	return 0;

}

File: Assignment1/fourargs.py
#copyright 2018 Zhuyu Li zhuyuli@bu.edu
"""
A python program that outputs the first four arguments to
stdout and any additional arguments to stderr
"""

import sys

arg_length = len(sys.argv)

if arg_length > 5:
    for num in range(1, 5):
        sys.stdout.write(sys.argv[num] + '\n')

    for num in range(5, arg_length):
        sys.stderr.write(sys.argv[num] + '\n')
        
        
else:
    for num in range(1, arg_length):
        sys.stdout.write(sys.argv[num] + '\n')
        
              
     
    



